zapcore/field_test.go

```diff
@@ -58,6 +57,27 @@ func (u users) MarshalLogArray(enc ArrayEncoder) error {
 	return nil
 }
 
+type obj struct {
+	kind int
+}
+
+func (o *obj) String() string {
```

Comment:
let's add a custom object type which mimics production use-cases: we have a `String` method which accesses some field, but it causes a panic because the object is nil.

Reply:
There is the following additional test in line 103:
`{t: StringerType, iface: (*url.URL)(nil), want: empty, err: "runtime error: invalid memory address or nil pointer dereference"},`

Is that sufficient for you? The current obj struct makes testing some of the edge cases a lot simpler but I am happy to change that.

Reply:
That case seems sufficient; it exercises the `fmt.Stringer` with a nil value case.

Reply:
yep makes sense, thanks